src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/TrafficSeo/ShopUrlType.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Configure\ShopParameters\TrafficSeo;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * Class ShopUrlType is responsible for providing form fields for
+ * Shop parameters -> Traffic & Seo -> Seo & Urls -> Shop urls block.
+ */
+class ShopUrlType extends AbstractType
+{
+    /**
+     * @var bool
+     */
+    private $isHostMode;
+
+    /**
+     * @var bool
+     */
+    private $isShopFeatureActive;
+
+    /**
+     * @var bool
+     */
+    private $isMainShopUrlExists;
+
+    /**
+     * ShopUrlType constructor.
+     *
+     * @param bool $isHostMode
+     * @param bool $isShopFeatureActive
+     * @param bool $isMainShopUrlExists
+     */
+    public function __construct($isHostMode, $isShopFeatureActive, $isMainShopUrlExists)
+    {
+        $this->isHostMode = $isHostMode;
+        $this->isShopFeatureActive = $isShopFeatureActive;
+        $this->isMainShopUrlExists = $isMainShopUrlExists;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        if (!$this->isHostMode && !$this->isShopFeatureActive && $this->isMainShopUrlExists) {
```

Comment:
@sarjon - when these fields are not rendered the type title "Shop urls" is still visible when the forms are rendered. It is due to form_rest is included. Do you know the clean way of how I can avoid that? I can use `{{ form_end(form, {'render_rest': false}) }}` and render token manually in meta.html.twig but I don't believe its the best option :thinking: